### PR TITLE
Add mkdocs redirects

### LINKED ISSUE #21: Avoid using .html extension for pages
Project should be configured in such way that .html extensions are not visible to the user (nice urls), all being listed as "folders".

Example https://ansible.readthedocs.io/projects/dev-tools/

Any existing incoming links can be redirected using something like https://github.com/ansible/ansible-lint/blob/main/docs/redirects.yml

### PR DESCRIPTION
**Pull request by oraNod**
This PR adds [`mkdocs-redirects`](https://github.com/mkdocs/mkdocs-redirects) and sets up a redirect for the ecosystem page that was removed with https://github.com/ansible/ecosystem-documentation/commit/f691e36bc45e99b2e3893d2b85ab0bd7a7e8e118

Reading through the mkdocs-redirects documentation it seems like it might be a good idea to fix https://github.com/ansible/ecosystem-documentation/issues/21 before we merge this. I'll send a separate PR.

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ def pip_compile(session: nox.Session, req: str):
     # Use --upgrade by default unless a user passes -P.
     args = list(session.posargs)
     if not any(
-        arg.startswith("-P") or arg.startswith("--upgrade-package") for arg in args
+        arg.startswith(("-P", "--upgrade-package", "--no-upgrade")) for arg in args
     ):
```
Comment (samccann) [inline noxfile.py]: Is this related to this pr? I'm fuzzy on nox stuff but not immediately seeing how this relates to the mkdocs redirects.

Comment (oraNod) [reply noxfile.py]: Yeah it probably should go into a separate PR. I had to add the `--no-upgrade` flag here so that only the `mkdocs-redirects` package would get added to the requirements. Otherwise the nox session upgrades all the other packages.

Comment (oraNod) [reply noxfile.py]: Moved to https://github.com/ansible/ecosystem-documentation/pull/32

**Review by samccann — APPROVED** [review]
Verified by typing https://ansible--28.org.readthedocs.build/en/28/ecosystem.html and seeing that it does redirect to the ecosystem page.

**Review by ssbarnea — APPROVED** [review]
